### PR TITLE
Fixed issue #19139: CSRF in Reset Survey menus 

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/controllers/admin/SurveymenuController.php [hunk 1]
```diff
@@ -21,6 +21,7 @@ public function __construct($controller, $id)
 
     /**
      * @return string[] action filters
+     * @todo: Review if this code is actually running.
```
Comment (Shnoulle) [inline application/controllers/admin/SurveymenuController.php:24]: :)

File: application/controllers/admin/SurveymenuController.php [hunk 2]
```diff
@@ -284,52 +285,27 @@ public function delete()
...
     {
+        $this->requirePostRequest();
+
         if (!(Permission::model()->hasGlobalPermission('settings', 'delete') && Permission::model()->hasGlobalPermission('settings', 'update'))) {
             Yii::app()->user->setFlash('error', gT("Access denied"));
             $this->getController()->redirect(Yii::app()->createUrl('/admin'));
         }
 
         $model = Surveymenu::model();
-        if (Yii::app()->request->isPostRequest) {
```
Comment (Shnoulle) [inline application/controllers/admin/SurveymenuController.php:293]: This part didn't force POST ? We have surely some other place where it's like this, no ?

File: application/controllers/admin/SurveymenuController.php [hunk 3]
```diff
@@ -284,52 +285,27 @@ public function delete()
...
 
-                return Yii::app()->getController()->renderPartial(
-                    '/admin/super/_renderJson',
-                    $returnData,
-                    false,
-                    false
-                );
-            }
-            //get model to do the work
-            $success = $model->restoreDefaults();
```
Comment (Shnoulle) [inline application/controllers/admin/SurveymenuController.php:322]: Oh ! no get it ! If it's not POST : it get there …

**Review by Shnoulle — APPROVED** [review]

**Comment by gabrieljenik** [comment]
Found some more stuff to fix.
Something is off with the noCRSFrules.
Will update shortly

**Pull request merged**